File: src/main/java/org/tvl/tvlooker/domain/model/Review.java
package org.tvl.tvlooker.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.sql.Timestamp;
import java.util.UUID;

@Builder
@Getter
@AllArgsConstructor
public class Review {
    private final Long id;
    private final UUID userId;
    private final Long itemId;
    private final Integer score;
    @Builder.Default
    private final String reviewText = "";
    private final Timestamp reviewDate;
}

File: frontend/src/types/review.ts
import type { UUID } from "crypto";

export type Review = {
  id: number;
  userId: UUID;
  itemId: number;
  rating: number; // 1-5
  comment: string;
  createdAt: string;
};

export type CreateReviewRequest = {
  userId: UUID;
  itemId: number;
  rating: number;
  comment: string;
}

export type UpdateReviewRequest = {
  score?: number;
  reviewText?: string;
};
